lib/Dialect/Torch/Transforms/AdjustCallingConventions.cpp

```diff
@@ -186,53 +187,11 @@ class AdjustCallingConventionForReturn
 };
 } // namespace
 
-static bool isValidNonContainerResultType(Type resultType) {
-  return resultType.isa<Torch::BaseTensorType>() ||
-         resultType.isa<Torch::FloatType>() ||
-         resultType.isa<Torch::IntType>() ||
-         resultType.isa<Torch::BoolType>() ||
-         resultType.isa<Torch::NoneType>();
-}
-
-static LogicalResult validateReturns(func::FuncOp func) {
-  if (func.getResultTypes().size() > 1) {
-    return func->emitError(
-      "Functions directly imported from Python should only ever return one "
-      "item. Multiple return values are returned as a tuple.");
-  }
-
-  // Allow returns of nothing. This shouldn't be possible from Python, but it
-  // can happen in IR that's been directly constructed.
-  if (func.getResultTypes().size() == 0)
-    return success();
-
-  const auto& resultType = func.getResultTypes().front();
-
-  // Allow single tensor, scalar, and bool returns
-  if (isValidNonContainerResultType(resultType)) {
-    return success();
-  }
-
-  // Allow multi-tensor/scalar/bool tuple returns
-  if (auto tuple = resultType.dyn_cast<Torch::TupleType>()) {
-    const auto& containedTypes = tuple.getContainedTypes();
-    bool containsValidTypes = llvm::all_of(
-      tuple.getContainedTypes(), isValidNonContainerResultType);
-    if (containedTypes.size() >= 2 && containsValidTypes) {
-      return success();
-    }
-  }
-
-  return func->emitError(
-    "Functions must return a single tensor-like value, multiple tensor-like "
-    "values, or a tuple of more than one tensor-like value. Tensor-like "
-    "values: tensors, scalars, bools, and Nones.");
-}
-
 static LogicalResult adjustCallingConventions(func::FuncOp func,
                                               TypeBoundMap &typeBoundMap) {
-  if (failed(validateReturns(func)))
```

Comment:
This code is a remnant of when we used to only run once the pipeline 

https://github.com/llvm/torch-mlir/blob/5223f990dfff28a1927ef61d70f58f6da2fc9138/lib/Dialect/Torch/Transforms/Passes.cpp#L33

When that was the case, it was important for passes to verify success before finishing and moving on to the next pass. However, we now run the lowering pipeline over and over until a valid IR satisfying the backend contract is reached. This means that we should remove the verification from these passes and move it to 

https://github.com/llvm/torch-mlir/blob/5223f990dfff28a1927ef61d70f58f6da2fc9138/lib/Dialect/Torch/Transforms/LowerToBackendContract.cpp#L176

Reply:
Ok, done.
Looking at the code with your comment as context, it looks like that code can actually be removed.
The current backend contract already allows what we care about, and has clear explanations on the different expectations.

I added a unit test in particular for what we care about in this PR, plus a few extra on the way.